### ISSUE TITLE
Same loss values but different eval result

### ISSUE THREAD

**Issue by a-maci** [huggingface/pytorch-pretrained-BERT#516]
I am experimenting with low-precision on the pre-trained BERT for SQuAD scenario. 
I am seeing a strange issue: the loss value when fine-tuning the model with FP16 is very similar to the loss value when fine-tuning the model at Int8. However, the eval results are are quite different -- with Int8, the results are quite bad (f1 = 3) compared to f1=88 with FP16.

Any idea what is going on and suggestions for debugging?

**Comment by thomwolf** [comment]
I have never tried Int8 in PyTorch.
Can you share some code so we can have a look?
